tools/wptrunner/wptrunner/testloader.py

```diff
@@ -31,29 +33,113 @@ def do_delayed_imports():
     from manifest.download import download_from_github  # type: ignore
 
 
-class TestGroupsFile:
-    """
-    Mapping object representing {group name: [test ids]}
-    """
-
-    def __init__(self, logger, path):
+class TestGroups:
+    def __init__(self, logger, path, subsuites):
         try:
             with open(path) as f:
-                self._data = json.load(f)
+                data = json.load(f)
         except ValueError:
             logger.critical("test groups file %s not valid json" % path)
             raise
 
+        self.tests_by_group = defaultdict(set)
         self.group_by_test = {}
-        for group, test_ids in self._data.items():
+        for group, test_ids in data.items():
             for test_id in test_ids:
-                self.group_by_test[test_id] = group
+                id_parts = group.split(":", 1)
+                if len(id_parts) == 1:
+                    group_name = id_parts[0]
+                    subsuite = None
+                else:
+                    subsuites, group_name = id_parts
+                    if subsuite not in subsuites:
+                        raise ValueError(f"Unknown subsuite {subsuite} in group data {group}")
```

Comment:
Maybe I misunderstand the RFC, but shouldn't this just be:

```suggestion
                    subsuite, group_name = id_parts
```

As is, it seems that `subsuite` can be undefined if we never took the other branch, as for:

```json
{"my-subsuite:my-group": [...]}
```

Also, I think we can lift the assignments to `subsuite, group_name` out of the inner loop, since that shouldn't change across test IDs.